### PR TITLE
Allow editing of join table attributes

### LINKED ISSUE #31: Support for attributes on "in-between" tables of n:m relations
We would like to enhance the linking relation editor to support attributes on the in-between table of a n:m relation.

Our use case is to assign fauna and flora observations to "Beurteilungsformulare" (assessment forms), and each observation needs to have 1-2 additional attributes, such as nr of species sighted, distribution, etc.

One option would be to have a "tree-view"-like widget where one could add the additional values.

### PR DESCRIPTION
**Pull request by domi4484**
Allow editing of attributes in the join table feature thanks to an expandable QTreeWidgetItem that shows the attribute form.

Still missing:
- [ ] Attributes not editable when linking new feature
- [ ] Documentation
- [ ] Tests

Fix #31 

image

A new configuration was added to enable/disable this functionality
image

### CODE REVIEW

**Comment by domi4484** [comment]
@signedav , it is not finished, but maybe you can already take a look. 
If I find the time tomorrow before going in holiday, I will implement the missing part of editing attributes of new linked features.
In each case, even if still quite raw, I would like to make an experimental release tomorrow, to gather for an early feedback by @andreasneumann.

I had to replace the ListView/Model approach with a TreeWidget/Items because it was to complicated to handle custom widgets in the model.

**Comment by signedav** [comment]
Thanks. I cannot make a proper review until earliest Tuesday but feel free to make the experimental (should be possible on this unmerged branch - otherwise merge it). I will have a closer look at the code later this week.

**Comment by domi4484** [comment]
oh thanks i did not thought about releasing from this branch. I will do it like that

**Comment by signedav** [comment]
Never did it before btw. but I guess it should work...

**Comment by domi4484** [comment]
I'll merge it to have also the multi-word filter in the pre-release

File: linking_relation_editor/gui/linking_relation_editor_config_widget.py [hunk 1]
```diff
@@ -37,21 +45,42 @@ def __init__(self, relation, parent):
...
+            if layout.objectName() == formLayoutName:
+                formLayout = layout
+                break
 
+        if formLayout is None:
+            QgsLogger.warning(self.tr("QFormLayout with object name '{}' not found".format(formLayout)))
+        else:
+            self.linkingChildManagerDialogConfigWidget = LinkingChildManagerDialogConfigWidget()
+
+            formLayout.insertRow(5, self.linkingChildManagerDialogConfigWidget)
```
Comment (signedav) [inline linking_relation_editor/gui/linking_relation_editor_config_widget.py:63]: I think it's inserting it again and again on every change from relation editor to linking relation editor.

File: linking_relation_editor/gui/linking_child_manager_dialog.py [hunk 2]
```diff
@@ -47,6 +50,7 @@ def __init__(
         nmRelation: QgsRelation,
         editorContext: QgsAttributeEditorContext,
         oneToOne: bool,
+        config: dict,
```
Comment (signedav) [inline linking_relation_editor/gui/linking_child_manager_dialog.py:53]: This is the linking child dialog config right? could this be named more descriptive?

**Review by signedav — COMMENTED** [review]
Hi Damiano

It looks good. I will have another look and make further tests when it's finalized.

File: linking_relation_editor/gui/linking_relation_editor_config_widget.py (hunk 1 above)
Comment (domi4484) [reply linking_relation_editor/gui/linking_relation_editor_config_widget.py:63]: This is in the constructor, so it should happen only once

File: linking_relation_editor/gui/linking_child_manager_dialog.py (hunk 2 above)
Comment (domi4484) [reply linking_relation_editor/gui/linking_child_manager_dialog.py:53]: Yes, and in fact normally I like long descriptive names :)
renamed in https://github.com/opengisch/qgis-linking-relation-editor/pull/41

**Pull request merged**